mesonbuild/interpreter/type_checking.py

```diff
@@ -347,3 +380,52 @@ def _output_validator(outputs: T.List[str]) -> T.Optional[str]:
     validator=in_set_validator({'set', 'prepend', 'append'}))
 
 ENV_SEPARATOR_KW = KwargInfo('separator', str, default=os.pathsep)
+
+DEPENDENCIES_KW: KwargInfo[T.List[Dependency]] = KwargInfo(
+    'dependencies',
+    ContainerTypeInfo(list, (Dependency, ExternalLibrary, InternalDependency)),
+    listify=True,
+    default=[],
+)
+
+link_with_error = '''can only be self-built targets, external dependencies (including libraries) must go in "dependencies".'''
```

Comment:
If this is a global constant, which it looks like it is, and probably only used in this file, preferably this should be `__LINK_WITH_ERROR`.

Reply:
Just `_LINK_WITH_ERROR` is fine. `__` instructus python to mangle the symbol in a way that prevents collissions in class heirarchies

Reply:
It's a two-functions-local constant, really. :)

Feels weird to name a variable in shouty-caps when its purpose is not to be "read-only, available to the entire file" but rather "the same string is used in two places 8 lines apart and I didn't want to type it twice, but it should go out of brain-scope as soon as it scrolls off the screen".